src/components/VisaProgressBar.tsx

```diff
@@ -0,0 +1,45 @@
+import {
+  VisaProgress,
+  FormConfig,
+  SectionName,
+  getOverallPromptIndex,
+} from '@lib/domain/form'
+import { Qualifications } from '@lib/domain/qualifications'
+import { isPromptCompleted } from '@lib/domain/prompts'
+export function VisaProgressBar({
+  config,
+  qualifications,
+}: {
+  config: FormConfig
+  qualifications: Qualifications
+}) {
+  const totalPrompts = Object.values(config.sections).reduce(
+    (accumulator, value) => accumulator + value.length,
+    0,
+  )
+  const sections: SectionName[] = config.order
+  let progress: number = 0
+
+  sections.map(section => {
+    const startPromptId = getOverallPromptIndex(config, section, 0)
+    const prompts = config.sections[section]?.length ?? 0
+    const promptIdsToCheck = [...Array(prompts)].map(
+      (_, i) => i + startPromptId,
+    )
+    promptIdsToCheck.map(promptId => {
```

Comment:
I think we can simplify this because we don't need to worry about what section a prompt belongs to; we already have the total number of prompts available.

This logic is also a good candidate to be extracted out to a function that can be tested separately.

Reply:
i think i kinda get your point here 🤔. i reckon we can just loop through all the ids since they're deterministic (0 --> however many there are) so re wrote it to simplify that